# Experiment 1: Estimating Apple's CAPM Expected Return Relative to the S&P 500


**Research Question:** Using monthly return data, what expected return does CAPM imply for Apple stock based on its beta relative to the S&P 500? <br>
**Data:** <br>
    - Apple adjusted monthly prices <br>
    - SPY adjusted monthly prices as the market proxy <br>
    - Monthly risk-free rate, using the 1-month Treasury bill rate <br>
**Method:** <br>
    1. Calculate monthly returns for Apple and SPY. <br>
    2. Convert both series into excess returns by subtracting the monthly risk-free rate. <br>
    3. Estimate Apple’s beta using an OLS regression of Apple excess returns on market excess returns. <br>
    4. Calculate the historical market risk premium. <br>
    5. Use CAPM to estimate Apple’s expected return. <br>
    6. Compare CAPM-implied expected return with Apple’s realized average return. <br>
**Sample:** 5 years of data: 2019 - 2024

## Step 1: Import the Data

First, we run this in the terminal: <br>
<br>
python3 -m pip install yfinance pandas numpy statsmodels matplotlib <br> <br>

Then, we run the code below

In [4]:
import yfinance as yf
import pandas as pd
import numpy as np
import statsmodels.api as sm
import matplotlib.pyplot as plt

Matplotlib is building the font cache; this may take a moment.


Second, we run the following code. This gives us the closing prices for Apple and the S&P 500 (using SPY as a proxy) for the time frame specified

In [ ]:
tickers = ["AAPL", "SPY"] # Specific ticker symbol requested

prices = yf.download( #retrives the data from yfinance
    tickers,
    start="2019-01-01", # Starting date
    end="2024-01-01", # Ending date
    interval="1mo", # Interval of time we want to observe
    auto_adjust=True # Adjusts historical price for stock splits and dividends
)["Close"] # Specifies that we want the chosing price

prices.head

## Step 2: Calculate the Monthly Returns for APPL and SPY

In [7]:
returns = prices.pct_change().dropna()

returns.head() #Return only gives you the first 5 months. Also, since there is no change possible in Jan 2019, it is dropped.#

Ticker,AAPL,SPY
Date,,
2019-02-01,0.044777,0.032416
2019-03-01,0.097026,0.018101
2019-04-01,0.056436,0.040852
2019-05-01,-0.124213,-0.063771
2019-06-01,0.130519,0.069586


## Step 3: Calculate Beta

In [ ]:
y = returns["AAPL"] #Apple's monthly return
x = returns["SPY"] #SPY's monthly return

x = sm.add_constant(x) #This adds a column of 1s to the x data. This lets the regression estimate an intercept
model = sm.OLS(y, x).fit() #OLS is the Ordinary Least Squares. .fit is the thing that is actually doing the regression

print(model.summary()) #This just prints out the regression report

                            OLS Regression Results                            
Dep. Variable:                   AAPL   R-squared:                       0.642
Model:                            OLS   Adj. R-squared:                  0.635
Method:                 Least Squares   F-statistic:                     102.1
Date:                Wed, 08 Jul 2026   Prob (F-statistic):           2.56e-14
Time:                        20:48:01   Log-Likelihood:                 91.254
No. Observations:                  59   AIC:                            -178.5
Df Residuals:                      57   BIC:                            -174.4
Df Model:                           1                                         
Covariance Type:            nonrobust                                         
                 coef    std err          t      P>|t|      [0.025      0.975]
------------------------------------------------------------------------------
const          0.0144      0.007      2.052      0.0

**Interpreting the Results:** <br>
&nbsp;&nbsp;&nbsp;&nbsp; - &nbsp; Alpha / intercept: 0.0144 <br> 
&nbsp;&nbsp;&nbsp;&nbsp; - Beta on SPY: 1.3090 <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; - &nbsp; Apple tends to be more sensitive to market movements than SPY. Each increase in SPY's monthly return by 1 percentage points tends to increase Apple's by 1.31. <br>
&nbsp;&nbsp;&nbsp;&nbsp; - &nbsp; R-squared: 0.642 <br>
&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp; - &nbsp; About 64.2% of the variation in Apple's monthly returns is explained by SPY's monthly returns. <br>
&nbsp;&nbsp;&nbsp;&nbsp; - &nbsp; Observations: 59

**Important Note:** You can find the alpha, beta, and r_squared by themselves with the code below

In [ ]:
alpha = model.params["const"]
beta = model.params["SPY"]
r_squared = model.rsquared

print(f"Alpha: {alpha:.4f}") #The .4f is giving you it to the thousandths decimal place
print(f"Beta: {beta:.4f}")
print(f"R-squared: {r_squared: .4f}")

Alpha: 0.014
Beta: 1.3090
R-squared:  0.6418


# Alteration: Add Fama-French RF